ra/ra.go

```diff
@@ -554,20 +554,29 @@ func (ra *RegistrationAuthorityImpl) NewAuthorization(ctx context.Context, reque
 	if err != nil {
 		return core.Authorization{}, err
 	}
+
+	if features.Enabled(features.NewAuthorizationSchema) {
+		resultPB, err := ra.SA.NewAuthorization(ctx, authzPB)
+		if err != nil {
+			return core.Authorization{}, err
+		}
+		id := fmt.Sprintf("%d", *resultPB.Id)
+		authzPB.Id = &id
+		return bgrpc.PBToAuthz(authzPB)
```

Comment:
This is a bit odd - why are we returning `authzPB` (the input to `SA.NewAuthorization, plus a copied ID field) instead of `resultPB`? At a minimum there should be a comment explaining this, but if we can return the `resultPB` that would be better.

Reply:
`resultPB` is a `sapb.AuthorizationID2`, which is just the ID.